src/app/models/User.ts

```diff
@@ -180,6 +178,7 @@ export interface ISessionUser extends IUser {
  * A user model for the website user
  */
 export class SessionUser extends User implements ISessionUser {
+  // ! All fields are persisted because model is saved to, and read from, localstorage
```

Comment:
I have tried to reduce the stored information already to what i suspect will potentially be needed across the site. Here are the values stored and why:

- `id` => Self explanatory
- `email` => Should be removed
- `userName` => Used by header and menu links
- `imageUrls` => Used by header (could instead just request the User data in the header, but why duplicate requests?
- `preferences` => I had assumed system wide preferences were here and would potentially be needed in future
- `rolesMask` => Needed for menus
- `rolesMaskNames` => Should be removed (have rolesMask already)
- `timezoneInformation` => I had assumed this would be used system wide somewhere in future such as automatically converting times or time formatting
- `authToken` => Self explanatory

Reply:
Revision:
`User` model output:

- id
- email
- userName
- imageUrls
- preferences
- rolesMask
- timezoneInformation

`SessionUser` model output:

- id
- authToken
- userName
- imageUrls
- preferences
- rolesMask
- timezoneInformation

Reply:
> * userName
> * imageUrls
> * preferences
> * rolesMask
> * timezoneInformation
> 
> `SessionUser` model output:
> 
> * id
> * authToken
> * userName
> * imageUrls
> * preferences
> * rolesMask
> * timezoneInformation

Okay, the proposed model seems fine. Could we see a unit test that asserts *only* the required properties are saved in SessionUser?

I still have an issue with `SessionUser` extending `User`. User has a bunch of extra fields/properties that will never be populated for a SessionUser.

Reply:
No need to add the assertion, I will use the `@BawPersistAttr` to create the `toJSON()` functions

Reply:
Normally, correct, yes, it would not need to be tested - we don't need to reproduce the tests for `@BawPersistAttr`. In this case however, I think it is important we don't store potentially sensitive information so we should test for that explicitly. Different use-case even if testing same fields.

Reply:
I already have a test for this, I will update the test name to increase clarity.